tests/pds/dataservice_test.go

```diff
@@ -614,3 +617,437 @@ func UpgradeDataService(dataservice, oldVersion, oldImage, dsVersion, dsBuild st
+
+				defer func() {
+					if !isDeploymentsDeleted {
+						Step("Delete created deployments")
+						resp, err := pdslib.DeleteDeployment(deployment.GetId())
+						Expect(err).NotTo(HaveOccurred())
+						Expect(resp.StatusCode).Should(BeEquivalentTo(http.StatusAccepted))
+					}
+				}()
+
+				Step("Validate Storage Configurations", func() {
+
+					logrus.Infof("data service deployed %v ", ds)
+					dash.Infof("data service deployed %v ", ds)
+					resourceTemp, storageOp, config, err := pdslib.ValidateDataServiceVolumes(deployment, ds.Name, dataServiceDefaultResourceTemplateID, storageTemplateID, nname)
+					Expect(err).NotTo(HaveOccurred())
+					logrus.Infof("filesystem used %v ", config.Spec.StorageOptions.Filesystem)
+					logrus.Infof("storage replicas used %v ", config.Spec.StorageOptions.Replicas)
+					logrus.Infof("cpu requests used %v ", config.Spec.Resources.Requests.CPU)
+					logrus.Infof("memory requests used %v ", config.Spec.Resources.Requests.Memory)
+					logrus.Infof("storage requests used %v ", config.Spec.Resources.Requests.Storage)
+					logrus.Infof("No of nodes requested %v ", config.Spec.Nodes)
+					logrus.Infof("volume group %v ", storageOp.VolumeGroup)
+
+					Expect(resourceTemp.Resources.Requests.CPU).Should(Equal(config.Spec.Resources.Requests.CPU))
+					Expect(resourceTemp.Resources.Requests.Memory).Should(Equal(config.Spec.Resources.Requests.Memory))
+					Expect(resourceTemp.Resources.Requests.Storage).Should(Equal(config.Spec.Resources.Requests.Storage))
+					Expect(resourceTemp.Resources.Limits.CPU).Should(Equal(config.Spec.Resources.Limits.CPU))
+					Expect(resourceTemp.Resources.Limits.Memory).Should(Equal(config.Spec.Resources.Limits.Memory))
+					repl, err := strconv.Atoi(config.Spec.StorageOptions.Replicas)
+					Expect(err).NotTo(HaveOccurred())
+					Expect(storageOp.Replicas).Should(Equal(int32(repl)))
+					Expect(storageOp.Filesystem).Should(Equal(config.Spec.StorageOptions.Filesystem))
+					Expect(config.Spec.Nodes).Should(Equal(int32(ds.Replicas)))
+					cleanup = append(cleanup, deployment)
+
+				})
+
+			}
+
+			logrus.Infof("List of created deployments: %v ", cleanup)
+			dash.Infof("List of created deployments: %v ", cleanup)
+
+			Step("Delete created namespace", func() {
+				logrus.Infof("Cleanup: Deleting created namespace %v", nname)
+				dash.Infof("Cleanup: Deleting created namespace %v", nname)
+				err := pdslib.DeleteK8sPDSNamespace(nname)
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			Step("Verify that the namespace was deleted", func() {
+				err := pdslib.ValidateK8sNamespaceDeleted(nname)
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			Step("Verify created deployments have been deleted", func() {
+				for _, dep := range cleanup {
+					err := pdslib.ValidateDataServiceDeploymentNegative(dep, nname)
+					Expect(err).NotTo(HaveOccurred())
+				}
+				isDeploymentsDeleted = true
+			})
+
+		})
+
+	})
+})
+
+var _ = Describe("{RestartPXPods}", func() {
```

Comment:
same as above, include before and after each methods

Reply:
done